pkg/storage/storage.go

```diff
@@ -163,6 +164,49 @@ func New(cfg *config.Config) (*Storage, error) {
 		return tree.New()
 	}
 
+	// load the total memory of the server
+	vm, err := mem.VirtualMemory()
+	if err != nil {
+		return nil, err
+	}
+
+	interval := time.Duration(10)
+	// start a timer for checking if the memory used by application
+	// is more than 25% of the total memory, if so, trigger eviction
+	// with 10% to every cache
+	go func() {
```

Comment:
Perhaps it would be better to have this as a separate function.

Reply:
good idea